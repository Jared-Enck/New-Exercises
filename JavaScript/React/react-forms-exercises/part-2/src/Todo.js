import React from 'react'

function Todo({id, text, remove}) {
    const capsFirstLetters = (str) => {
        return str.toLowerCase()
            .split(' ')
            .map((s) => s.charAt(0).toUpperCase() + s.substring(1))
            .join(' ');
    }
    const newText = capsFirstLetters(text)
    return (
        <li className='d-flex flex-grow list-group-item bg-light'>
            <span className='flex-grow-1'>
                {newText}
            </span>
            <button 
                onClick={() => remove(id)}
                className='btn btn-sm btn-danger'
            >
                X
            </button>
        </li>
    )
}

export default Todo