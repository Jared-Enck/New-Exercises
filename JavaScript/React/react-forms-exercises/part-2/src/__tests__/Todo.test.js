import React from "react";
import { render } from "@testing-library/react";
import Todo from "../Todo";

const str = 'test todo'

it("renders without crashing", function() {
    render(<Todo text={str} />);
});
  
it("matches snapshot", function() {
    const {asFragment} = render(<Todo text={str} />);
    expect(asFragment()).toMatchSnapshot();
});