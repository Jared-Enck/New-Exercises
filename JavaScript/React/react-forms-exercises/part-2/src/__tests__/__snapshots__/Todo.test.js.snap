// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="d-flex flex-grow list-group-item bg-light"
  >
    <span
      class="flex-grow-1"
    >
      Test Todo
    </span>
    <button
      class="btn btn-sm btn-danger"
    >
      X
    </button>
  </li>
</DocumentFragment>
`;
